packages/runtime/client/rest/src/contract/endpoints/sources.ts

```diff
@@ -40,3 +40,94 @@ export const createContractSources = (axiosInstance: AxiosInstance) => {
     );
   };
 };
+
+export interface GetContractBySourceIdRequest {
+  contractSourceId: BuiltinByteString;
+  expand?: boolean;
+}
+
+export type GetContractBySourceIdResponse = Contract;
+
+const GetContractBySourceIdResponseGuard: t.Type<GetContractBySourceIdResponse> =
+  G.Contract;
+
+export const getContractSourceById =
+  (axiosInstance: AxiosInstance) =>
+  async ({
+    contractSourceId,
+    expand,
+  }: GetContractBySourceIdRequest): Promise<GetContractBySourceIdResponse> => {
+    const response = await axiosInstance.get(
+      `/contracts/sources/${encodeURIComponent(contractSourceId)}`,
+      { params: { expand } }
+    );
+    return pipe(
+      GetContractBySourceIdResponseGuard.decode(response.data),
+      E.match(
+        (e) => {
+          throw formatValidationErrors(e);
+        },
+        (e) => e
+      )
+    );
+  };
+
+export interface GetContractSourceAdjacencyRequest {
+  contractSourceId: BuiltinByteString;
+}
+
+export interface GetContractSourceAdjacencyResponse {
+  results: BuiltinByteString[];
```

Comment:
This shows the importance of the type alias

Reply:
Fair point.